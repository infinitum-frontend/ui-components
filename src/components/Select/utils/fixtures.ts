// export const SelectBaseOptions = [
//   { label: 'Депозитарные услуги', value: 'one' },
//   { label: 'Очень длинные спецдепозитарные услуги', value: 'two' },
//   { label: 'Консалтинг и аутсорсинг', value: 'three' }
// ]

export const SelectBaseOptions = [
  { label: 'Депозитарные услуги', value: 0 },
  { label: 'Спецдепозитарные услуги', value: 1 },
  { label: 'Консалтинг и аутсорсинг', value: 2 }
]

export const SelectBaseGroupedOptions = [
  { label: 'Депозитарные услуги', value: 'one' },
  {
    label: 'Спецдепозитарные услуги',
    options: [
      { label: 'Спецдепозитарные услуги 1', value: 'two-one' },
      { label: 'Очень длинные спецдепозитарные услуги 2', value: 'two-two' },
      { label: 'Спецдепозитарные услуги 3', value: 'two-three' }
    ]
  },
  {
    label: 'Консалтинг и аутсорсинг',
    options: [{ label: 'Консалтинг и аутсорсинг 1', value: 'three-one' }]
  }
]

export const SelectLongOptions = [
  { label: 'Депозитарные услуги', value: 'one' },
  { label: 'Очень длинные спецдепозитарные услуги', value: 'two' },
  { label: 'Консалтинг и аутсорсинг', value: 'three' },
  { label: 'Консалтинг и аутсорсинг 2', value: 'four' },
  { label: 'Консалтинг и аутсорсинг 3', value: 'five' },
  { label: 'Консалтинг и аутсорсинг 4', value: 'six' },
  { label: 'Консалтинг и аутсорсинг 5', value: 'sevent' },
  { label: 'Консалтинг и аутсорсинг 6', value: 'eight' },
  { label: 'Консалтинг и аутсорсинг 7', value: 'nine' },
  { label: 'Консалтинг и аутсорсинг 8', value: 'ten' }
]

export const SelectOptionsUnformatted = [
  { text: 'Депозитарные услуги', id: 0 },
  { text: 'Спецдепозитарные услуги', id: 1 },
  { text: 'Консалтинг и аутсорсинг', id: 2 }
]

export const SelectOptionsWithExtraKeys = [
  { text: 'Депозитарные услуги', id: 0, subtext: 'Subtext' },
  { text: 'Спецдепозитарные услуги', id: 1, subtext: 'Subtext' },
  { text: 'Консалтинг и аутсорсинг', id: 2, subtext: 'Subtext' }
]

export const SelectOptionsRawLong = [
  {
    id: '1',
    name: 'Ошибка доставки'
  },
  {
    id: '2',
    name: 'На отправке'
  },
  {
    id: '3',
    name: 'Доставлен'
  },
  {
    id: '4',
    name: 'Получен'
  },
  {
    id: '5',
    name: 'Ошибка в обработке'
  },
  {
    id: '6',
    name: 'Отправлен'
  },
  {
    id: '7',
    name: 'Отказано в обработке'
  },
  {
    id: '8',
    name: 'Сохранено'
  }
]
