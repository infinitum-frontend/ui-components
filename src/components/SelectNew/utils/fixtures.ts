export const SelectBaseOptions = [
  { label: 'Депозитарные услуги', value: 'one' },
  { label: 'Очень длинные спецдепозитарные услуги', value: 'two' },
  { label: 'Консалтинг и аутсорсинг', value: 'three' }
]

export const SelectBaseGroupedOptions = [
  { label: 'Депозитарные услуги', value: 'one' },
  {
    label: 'Спецдепозитарные услуги',
    options: [
      { label: 'Спецдепозитарные услуги 1', value: 'two-one' },
      { label: 'Очень длинные спецдепозитарные услуги 2', value: 'two-two' },
      { label: 'Спецдепозитарные услуги 3', value: 'two-three' }
    ]
  },
  {
    label: 'Консалтинг и аутсорсинг',
    options: [{ label: 'Консалтинг и аутсорсинг 1', value: 'three-one' }]
  },
  { label: 'Другое', value: 'four' }
]

export const SelectLongOptions = [
  { label: 'Депозитарные услуги', value: 'one' },
  { label: 'Очень длинные спецдепозитарные услуги', value: 'two' },
  { label: 'Консалтинг и аутсорсинг', value: 'three' },
  { label: 'Консалтинг и аутсорсинг 2', value: 'four' },
  { label: 'Консалтинг и аутсорсинг 3', value: 'five' },
  { label: 'Консалтинг и аутсорсинг 4', value: 'six' },
  { label: 'Консалтинг и аутсорсинг 5', value: 'sevent' },
  { label: 'Консалтинг и аутсорсинг 6', value: 'eight' },
  { label: 'Консалтинг и аутсорсинг 7', value: 'nine' },
  { label: 'Консалтинг и аутсорсинг 8', value: 'ten' }
]
