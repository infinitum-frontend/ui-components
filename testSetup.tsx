import { expect, beforeAll } from 'vitest'
import { render, RenderOptions } from '@testing-library/react'
// import '@testing-library/jest-dom/vitest' for vitest 1.x.x
import { ReactElement, JSXElementConstructor } from 'react'
import matchers from '@testing-library/jest-dom/matchers'

// extends Vitest's expect method with methods from react-testing-library
// https://github.com/testing-library/jest-dom#custom-matchers
expect.extend(matchers)

export const renderComponent = function (
  Component: ReactElement,
  options?: RenderOptions
): {
  el: HTMLElement
  container: HTMLElement
  rerender: (ui: ReactElement<any, string | JSXElementConstructor<any>>) => void
} {
  const { container, rerender } = render(Component, options)

  const el = container.firstChild as HTMLElement

  return { el, container, rerender }
}

beforeAll(() => {
  global.ResizeObserver = class ResizeObserver {
    observe(): any {}
    unobserve(): any {}
    disconnect(): any {}
  }

  // https://github.com/testing-library/react-testing-library/issues/95#issuecomment-392879866
  // https://github.com/vuejs/vue-test-utils/issues/319
  if (!HTMLElement.prototype.scrollTo) {
    HTMLElement.prototype.scrollTo = () => {}
  }
})

// runs a cleanup after each test case (e.g. clearing jsdom)
// afterEach(() => {
//   cleanup()
// })
