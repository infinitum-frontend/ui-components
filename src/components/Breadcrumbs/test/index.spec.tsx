import { it, describe } from 'vitest'
import { renderComponent } from '../../../../testSetup'
import { Breadcrumbs } from '../index'
import { screen } from '@testing-library/dom'
import { BreadcrumbsBaseItems, BreadcrumbsLongItems } from '../fixtures'
import userEvent from '@testing-library/user-event'

const user = userEvent.setup()

describe('Breadcrumbs', () => {
  it('should match snapshot', () => {
    const { el } = renderComponent(
      <Breadcrumbs items={BreadcrumbsBaseItems} maxVisibleCount={3} />
    )
    expect(el).toMatchSnapshot()
  })

  it('should have success render if the number of items is less than maxVisibleCount', () => {
    const { el } = renderComponent(
      <Breadcrumbs items={BreadcrumbsBaseItems} maxVisibleCount={10} />
    )
    expect(el).toMatchSnapshot('should match snapshot')
  })

  it('should isolate the interaction with last item', () => {
    renderComponent(
      <Breadcrumbs items={BreadcrumbsBaseItems} maxVisibleCount={3} />
    )
    const lastItem = screen.queryByText(
      BreadcrumbsBaseItems[BreadcrumbsBaseItems.length - 1].title
    ) as HTMLElement
    expect(lastItem).toBeInTheDocument()

    expect(lastItem).toHaveStyle('pointer-events: none')
  })
})

describe('Breadcrumbs Basic', () => {
  it('should match snapshot', () => {
    const { el } = renderComponent(
      <Breadcrumbs>
        <Breadcrumbs.Item as="a" href="https://specdep.ru/">
          Главная
        </Breadcrumbs.Item>
        <Breadcrumbs.Item as="a" href="https://specdep.ru/" target="_blank">
          Павел Прасс
        </Breadcrumbs.Item>
        <Breadcrumbs.Item>Хальмеев Александр</Breadcrumbs.Item>
      </Breadcrumbs>
    )
    expect(el).toMatchSnapshot()
  })
})

describe('Breadcrumbs With Dropdown', () => {
  it('should match snapshot', () => {
    const { el } = renderComponent(
      <Breadcrumbs items={BreadcrumbsLongItems} maxVisibleCount={3} />
    )
    expect(el).toMatchSnapshot()
  })

  it('should render button and open a dropdown menu', async () => {
    renderComponent(
      <Breadcrumbs items={BreadcrumbsLongItems} maxVisibleCount={3} />
    )
    const button = screen.queryByText('...') as HTMLElement
    expect(button).toBeInTheDocument()

    await user.click(button)

    const menu = screen.queryByRole('menu')
    expect(menu).toBeInTheDocument()
  })

  it('should isolate the interaction with last item', () => {
    renderComponent(
      <Breadcrumbs items={BreadcrumbsLongItems} maxVisibleCount={3} />
    )
    const lastItem = screen.queryByText(
      BreadcrumbsLongItems[BreadcrumbsLongItems.length - 1].title
    ) as HTMLElement
    expect(lastItem).toBeInTheDocument()

    expect(lastItem).toHaveStyle('pointer-events: none')
  })
})

describe('Breadcrumbs Dropdown Menu', () => {
  it('should match snapshot', async () => {
    renderComponent(
      <Breadcrumbs items={BreadcrumbsLongItems} maxVisibleCount={3} />
    )
    const button = screen.queryByText('...') as HTMLElement
    expect(button).toBeInTheDocument()

    await user.click(button)

    const menu = screen.queryByRole('menu')

    expect(menu).toMatchSnapshot('dropdown')
  })
})
