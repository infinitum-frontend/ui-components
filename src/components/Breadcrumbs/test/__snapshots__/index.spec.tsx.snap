// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Breadcrumbs > should have success render if the number of items is less than maxVisibleCount > should match snapshot 1`] = `
<nav
  aria-label="Breadcrumb"
  class="inf-breadcrumbs"
>
  <a
    class="inf-breadcrumbs-item"
    href="https://specdep.ru"
    title="Главная"
  >
    Главная
  </a>
  <svg
    class="inf-breadcrumbs-separator"
    fill="none"
    height="20"
    viewBox="0 0 20 20"
    width="20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8 15L13 10L8 5"
      stroke="currentColor"
      stroke-linejoin="round"
      stroke-width="1.5"
    />
  </svg>
  <a
    class="inf-breadcrumbs-item"
    href="https://specdep.ru"
    title="Прасс Павел"
  >
    Прасс Павел
  </a>
  <svg
    class="inf-breadcrumbs-separator"
    fill="none"
    height="20"
    viewBox="0 0 20 20"
    width="20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8 15L13 10L8 5"
      stroke="currentColor"
      stroke-linejoin="round"
      stroke-width="1.5"
    />
  </svg>
  <a
    class="inf-breadcrumbs-item"
    href="https://specdep.ru"
    title="Хальмеев Александр"
  >
    Хальмеев Александр
  </a>
</nav>
`;

exports[`Breadcrumbs > should match snapshot 1`] = `
<nav
  aria-label="Breadcrumb"
  class="inf-breadcrumbs"
>
  <a
    class="inf-breadcrumbs-item"
    href="https://specdep.ru"
    title="Главная"
  >
    Главная
  </a>
  <svg
    class="inf-breadcrumbs-separator"
    fill="none"
    height="20"
    viewBox="0 0 20 20"
    width="20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8 15L13 10L8 5"
      stroke="currentColor"
      stroke-linejoin="round"
      stroke-width="1.5"
    />
  </svg>
  <a
    class="inf-breadcrumbs-item"
    href="https://specdep.ru"
    title="Прасс Павел"
  >
    Прасс Павел
  </a>
  <svg
    class="inf-breadcrumbs-separator"
    fill="none"
    height="20"
    viewBox="0 0 20 20"
    width="20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8 15L13 10L8 5"
      stroke="currentColor"
      stroke-linejoin="round"
      stroke-width="1.5"
    />
  </svg>
  <a
    class="inf-breadcrumbs-item"
    href="https://specdep.ru"
    title="Хальмеев Александр"
  >
    Хальмеев Александр
  </a>
</nav>
`;

exports[`Breadcrumbs Basic > should match snapshot 1`] = `
<nav
  aria-label="Breadcrumb"
  class="inf-breadcrumbs"
>
  <a
    class="inf-breadcrumbs-item"
    href="https://specdep.ru/"
  >
    Главная
  </a>
  <svg
    class="inf-breadcrumbs-separator"
    fill="none"
    height="20"
    viewBox="0 0 20 20"
    width="20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8 15L13 10L8 5"
      stroke="currentColor"
      stroke-linejoin="round"
      stroke-width="1.5"
    />
  </svg>
  <a
    class="inf-breadcrumbs-item"
    href="https://specdep.ru/"
    target="_blank"
  >
    Павел Прасс
  </a>
  <svg
    class="inf-breadcrumbs-separator"
    fill="none"
    height="20"
    viewBox="0 0 20 20"
    width="20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8 15L13 10L8 5"
      stroke="currentColor"
      stroke-linejoin="round"
      stroke-width="1.5"
    />
  </svg>
  <span
    class="inf-breadcrumbs-item"
  >
    Хальмеев Александр
  </span>
</nav>
`;

exports[`Breadcrumbs Dropdown Menu > should match snapshot > dropdown 1`] = `
<div
  aria-labelledby="radix-:r8:"
  aria-orientation="vertical"
  class="inf-dropdown-menu-content"
  data-align="center"
  data-orientation="vertical"
  data-radix-menu-content=""
  data-side="bottom"
  data-state="open"
  dir="ltr"
  id="radix-:r9:"
  role="menu"
  style="outline: none; --radix-dropdown-menu-content-transform-origin: var(--radix-popper-transform-origin); --radix-dropdown-menu-content-available-width: var(--radix-popper-available-width); --radix-dropdown-menu-content-available-height: var(--radix-popper-available-height); --radix-dropdown-menu-trigger-width: var(--radix-popper-anchor-width); --radix-dropdown-menu-trigger-height: var(--radix-popper-anchor-height); pointer-events: auto;"
  tabindex="-1"
>
  <a
    as="a"
    class="inf-dropdown-menu-item"
    data-orientation="vertical"
    data-radix-collection-item=""
    href="https://specdep.ru"
    role="menuitem"
    tabindex="-1"
    title="Прасс Павел"
  >
    Прасс Павел
  </a>
  <a
    as="a"
    class="inf-dropdown-menu-item"
    data-orientation="vertical"
    data-radix-collection-item=""
    href="https://specdep.ru"
    role="menuitem"
    tabindex="-1"
    title="Хальмеев Александр"
  >
    Хальмеев Александр
  </a>
</div>
`;

exports[`Breadcrumbs Dropdown Menu > should match snapshot 1 > dropdown 1`] = `
<div
  data-side="bottom"
  data-align="center"
  role="menu"
  aria-orientation="vertical"
  data-state="open"
  data-radix-menu-content=""
  dir="ltr"
  id="radix-:r1:"
  aria-labelledby="radix-:r0:"
  class="inf-dropdown-menu-content"
  tabindex="-1"
  data-orientation="vertical"
  style="outline: none; --radix-dropdown-menu-content-transform-origin: var(--radix-popper-transform-origin); --radix-dropdown-menu-content-available-width: var(--radix-popper-available-width); --radix-dropdown-menu-content-available-height: var(--radix-popper-available-height); --radix-dropdown-menu-trigger-width: var(--radix-popper-anchor-width); --radix-dropdown-menu-trigger-height: var(--radix-popper-anchor-height); pointer-events: auto;"
>
  <a
    title="Прасс Павел"
    as="a"
    href="https://specdep.ru"
    role="menuitem"
    class="inf-dropdown-menu-item"
    tabindex="-1"
    data-orientation="vertical"
    data-radix-collection-item=""
  >
    Прасс Павел
  </a>
  <a
    title="Хальмеев Александр"
    as="a"
    href="https://specdep.ru"
    role="menuitem"
    class="inf-dropdown-menu-item"
    tabindex="-1"
    data-orientation="vertical"
    data-radix-collection-item=""
  >
    Хальмеев Александр
  </a>
</div>
`;

exports[`Breadcrumbs With Dropdown > should match snapshot 1`] = `
<nav
  aria-label="Breadcrumb"
  class="inf-breadcrumbs"
>
  <a
    class="inf-breadcrumbs-item"
    href="https://specdep.ru"
    title="Главная"
  >
    Главная
  </a>
  <svg
    class="inf-breadcrumbs-separator"
    fill="none"
    height="20"
    viewBox="0 0 20 20"
    width="20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8 15L13 10L8 5"
      stroke="currentColor"
      stroke-linejoin="round"
      stroke-width="1.5"
    />
  </svg>
  <button
    aria-expanded="false"
    aria-haspopup="menu"
    class="inf-breadcrumbs-show-more-button"
    data-state="closed"
    id="radix-:r0:"
    type="button"
  >
    ...
  </button>
  <svg
    class="inf-breadcrumbs-separator"
    fill="none"
    height="20"
    viewBox="0 0 20 20"
    width="20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8 15L13 10L8 5"
      stroke="currentColor"
      stroke-linejoin="round"
      stroke-width="1.5"
    />
  </svg>
  <a
    class="inf-breadcrumbs-item"
    href="https://specdep.ru"
    title="Леонид Хенкин"
  >
    Леонид Хенкин
  </a>
  <svg
    class="inf-breadcrumbs-separator"
    fill="none"
    height="20"
    viewBox="0 0 20 20"
    width="20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8 15L13 10L8 5"
      stroke="currentColor"
      stroke-linejoin="round"
      stroke-width="1.5"
    />
  </svg>
  <a
    class="inf-breadcrumbs-item"
    href="https://specdep.ru"
    title="Александр Круглов"
  >
    Александр Круглов
  </a>
</nav>
`;
